### DISCUSSION TITLE
middleware not working

### DISCUSSION THREAD

**Discussion by vadymchornyidev** [vercel/next.js#48658] [Help] [answered]
### Summary

middleware does not work at all, does not respond

versions:

`"next": "^13.3.0"`
`"typescript": "^5.0.4"`

folder structure
- .next
- node_modules
- pages
- middleware.ts

tscofig

```
{
  "compilerOptions": {
    "target": "es5",
    "lib": ["dom", "dom.iterable", "esnext"],
    "allowJs": true,
    "skipLibCheck": true,
    "strict": false,
    "forceConsistentCasingInFileNames": true,
    "noEmit": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "preserve",
    "incremental": true,
    "baseUrl": ".",
    "paths": {
      "@/*": ["./*"]
    }
  },
  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx", "middleware.ts"],
  "exclude": ["node_modules"]
}
```

middleware.ts

```
// middleware.ts

import { NextResponse } from 'next/server'
import type { NextRequest } from 'next/server'

export function middleware(request: NextRequest) {
    console.log('middleware.ts', request.nextUrl.pathname)

    if (request.nextUrl.pathname.startsWith('/about')) {
        return NextResponse.rewrite(new URL('/about-2', request.url))
    }

    if (request.nextUrl.pathname.startsWith('/dashboard')) {
        return NextResponse.rewrite(new URL('/dashboard/user', request.url))
    }
}
```

there are no logs in the console, redirects do not work.

<img width="279" alt="image" src="https://user-images.githubusercontent.com/65212516/233561105-009781f3-eb6a-40ea-94b6-72e0bb39c0c4.png">

**Comment by vadymchornyidev** [comment]
maybe its bug?

**Reply by vadymchornyidev** [reply]
@icyJoseph can you help me pls?

**Comment by icyJoseph** [comment]
Hi,

Do you have a repository to take a look at?

I just tested with a new project, with no `src` directory and `middleware.ts` at the root, and it works fine.

Make sure to remove `.next` and `node_modules`, run `npm install`, or which ever you are using, and try again.

**Reply by vadymchornyidev** [reply]
sure, I've sent an invite https://github.com/VadimChorrny/admin
after removing `.next` and `node_modules` and `yarn` nothing happened

**Comment by icyJoseph** [comment, marked as answer]
Hi,

@VadimChorrny the problem is that you are using `Custom Page Extension`:

> Changing these values affects all Next.js pages, including the following:
> 
> middleware.js
> pages/_document.js
> pages/_app.js
> pages/api/

- https://nextjs.org/docs/api-reference/next.config.js/custom-page-extensions

So you need to rename your `middleware.ts` to `middleware.page.ts`

**Reply by vadymchornyidev** [reply]
Yeah, man, thanks) 
It works! 🔥

**Reply by icyJoseph** [reply]
You can kick me off the repository now. Happy to help!
